src/org/mozartoz/truffle/nodes/call/TailCallThrowerNode.java

```diff
@@ -0,0 +1,28 @@
+package org.mozartoz.truffle.nodes.call;
+
+import org.mozartoz.truffle.nodes.DerefNode;
+import org.mozartoz.truffle.nodes.OzNode;
+import org.mozartoz.truffle.runtime.TailCallException;
+
+import com.oracle.truffle.api.dsl.CreateCast;
+import com.oracle.truffle.api.dsl.NodeChild;
+import com.oracle.truffle.api.dsl.NodeChildren;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+
+@NodeChildren({ @NodeChild("receiver"), @NodeChild("arguments") })
+public abstract class TailCallThrowerNode extends OzNode {
+
+	@CreateCast("receiver")
+	protected OzNode derefReceiver(OzNode var) {
+		return DerefNode.create(var);
+	}
+
+	public abstract Object executeTailCall(VirtualFrame frame, Object receiver, Object[] arguments);
```

Comment:
Do you mean the DerefNode or using the Truffle Specialization artillery? (yes, the deref should probably be done manually)